openedx/core/djangoapps/theming/template_loaders.py

```diff
@@ -35,6 +35,18 @@ def __init__(self, engine, dirs=None):
             self.dirs = theme_dirs + self.dirs
         super().__init__(engine, self.dirs)
 
+    def get_dirs(self):
+        """
+        Override get_dirs method.
+
+        Make the theme templates a priority, avoiding cashing templates for django ones.
+        """
+        self.dirs = self.engine.dirs
+        theme_dirs = self.get_theme_template_sources()
+        if isinstance(theme_dirs, list):
+            self.dirs = theme_dirs + self.dirs
+        return self.dirs
```

Comment:
I don't think we should be modifying `self.dirs` -- that will mean that we prepend the `theme_dirs` list every time we call this `get_dirs` method. Better to just assemble a `dirs` list in the method and return that.

```suggestion
            dirs = theme_dirs + dirs
        return dirs
```

Reply:
Thanks @pomegranited 
I'll try to check this myself soon.

Reply:
No worries @DmytroAlipov , just ping me whenever you're ready for a final review.
And I've pushed my suggestions to [this branch](https://github.com/openedx/edx-platform/compare/master...open-craft:edx-platform:jill/DmytroAlipov/override-get-dirs-method) in case that helps.

Reply:
Hi @DmytroAlipov , do you know roughly when you'll have a chance to look at this? Would be nice to get this fix into Redwood if we can.